python/erdos/streams.py

```diff
@@ -135,54 +196,100 @@ class LoopStream(object):
 
     Must call `set` on a ReadStream to complete the loop.
     """
-    def __init__(self):
+    def __init__(self, _name: str = None):
         self._py_loop_stream = PyLoopStream()
+        self._name = _name
+
+    @property
+    def name(self) -> Union[str, None]:
+        """ The name of the stream. (`None` if no name was given.) """
+        return self._name
 
-    def set(self, read_stream):
+    def set(self, read_stream: ReadStream):
+        logger.debug("Setting the read stream {} to the loop stream {}".format(
+            read_stream.name, self._name))
         self._py_loop_stream.set(read_stream._py_read_stream)
 
 
 class IngestStream(object):
-    """Used to send messages from outside of operators."""
-    def __init__(self):
-        self._py_ingest_stream = PyIngestStream(0)
+    """An :py:class:`IngestStream` enables drivers to inject data into a
+    running ERDOS application.
+    
+    The driver can initialize a new :py:class:`IngestStream` and connect it to
+    an :py:class:`Operator` through :py:func:`connect`. Similar to a 
+    :py:class:`WriteStream`, an :py:class:`IngestStream` provides a 
+    :py:func:`IngestStream.send` to enable the driver to send data to the 
+    operator to which it was connected.
+    """
+    def __init__(self, _name: Union[str, None] = None):
```

Comment:
Can you please add the name type annotation to the other constructors as well (e.g., WriteStream)?